cmd/cogs/optparse.go

```diff
@@ -17,28 +17,30 @@ func ifErr(err error) {
 	}
 }
 
-func getRawValue(cfgMap map[string]string) string {
+func getRawValue(cfgMap map[string]interface{}) string {
 	output := ""
 	// for now, no delimiter
 	for _, v := range cfgMap {
-		output = output + v
+		output = output + fmt.Sprintf("%s", v)
 	}
 	return output
 
 }
 
-func upperKeys(cfgMap map[string]string) map[string]string {
+// upperKeys should always return a flat associative array of strings
+// coercing any interface{} value into a string
+func upperKeys(cfgMap map[string]interface{}) map[string]string {
 	newCfgMap := make(map[string]string)
 	for k, v := range cfgMap {
-		newCfgMap[strings.ToUpper(k)] = v
+		newCfgMap[strings.ToUpper(k)] = fmt.Sprintf("%s", v)
```

Comment:
I wonder if there's a better pattern for coercing an `interface{}` into a `string` 🤔 ...